main.go

```diff
@@ -169,6 +173,21 @@ func main() {
 			"monitoring for ModelServing, please provide a monitoring namespace via the (--monitoring-namespace) flag.")
 	}
 
+	if enableMRInferenceServiceReconcile {
```

Comment:
Rather than setting up a reconciler from scratch, I wonder if it would be better to add a child-reconciler in the existing InferenceService reconciler, in this chunk:

https://github.com/opendatahub-io/odh-model-controller/blob/062439d7e7de72dcea37ba7238c94f29f1273abe/controllers/inferenceservice_controller.go#L89-L96